### ISSUE TITLE
torch.sigmoid backward loses a representable float64 derivative at large positive input

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#195681]
### 🐛 Describe the bug

`torch.sigmoid` returns `0.0` for a first-order derivative whose mathematically correct value is finite, normal, and nonzero in `float64`.

For `sigmoid(x)`, the derivative can be written stably as `exp(-x) / (1 + exp(-x))**2` for positive `x`. At `x = 37.42994775023705`, the high-precision reference derivative rounds to approximately `5.551115123125775e-17`, which is finite and representable in `float64`. However, PyTorch returns `0.0` for the gradient.

This issue reproduces on CPU, so CUDA availability does not affect the result.

```python
import torch

dtype = torch.float64

x = torch.tensor(37.42994775023705, dtype=dtype, requires_grad=True)

y = torch.sigmoid(x)
actual, = torch.autograd.grad(y, x)

expected = torch.tensor(5.551115123125775e-17, dtype=dtype)

print("actual:")
print(actual)
print("expected:")
print(expected)
```

### Actual vs expected result

```text
actual:
tensor(0., dtype=torch.float64)

expected:
tensor(5.5511e-17, dtype=torch.float64)
```

The discrepancy occurs in the first-order backward pass. The expected derivative is a finite, normal, nonzero `float64` value, but PyTorch collapses it to `0.0`.

### Versions

```text
PyTorch version: 2.12.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
Python version: 3.13.13 | packaged by Anaconda, Inc.
Python platform: Linux x86_64

Is CUDA available: False
GPU models:
  GPU 0: NVIDIA RTX 6000 Ada Generation
  GPU 1: NVIDIA RTX 6000 Ada Generation
Nvidia driver version: 570.211.01

CPU:
  Model name: AMD Ryzen Threadripper PRO 7985WX 64-Cores
  CPU(s): 128

Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[pip3] triton==3.7.0
```

cc @ezyang @albanD @gqchen @nikitaved @soulitzer @Varal7 @bobrenjc93
